### PR TITLE
Feature/backend kwargs json to netcdf

### LINKED ISSUE #286: Add backend_kwargs to cfgrib to_netcdf
In some cases we need additional backend _kwargs related to cfgrib in the xr.open_dataset command, for example to correctly handle multiple time dimensions of forecast data or to extract additional attributes from the grib file. This is not possible with current API of `cfgrib to_netcdf`.

I suggest an additional optional argument, `--backend-kwargs-json file.json`, where file.json  contains the backend kwargs to be used in xr.open_dataset

### PR DESCRIPTION
**Pull request by EddyCMWF**
Allow backend kwargs to be provided in the to_netcdf executable, either via a json format string, or a path to a json file.
Fixes #286

### CODE REVIEW

**Comment by EddyCMWF** [comment]
Hi @alexamici , can you review this PR which fixes issue #286

**Comment by alexamici** [comment]
@EddyCMWF very nice feature!

I have a few style comments:
- you need to format the code with `black` you may
- why do you move imports inside the code? this is a but confusing and both modules are non-optional

File: cfgrib/__main__.py
Comment (iainrussell) [inline cfgrib/__main__.py:80]: Hi @EddyCMWF, shouldn't this be **backend_kwargs**= json.load(f) here?

Comment (EddyCMWF) [reply cfgrib/__main__.py]: Thanks, was a typo from my refactoring

**Comment by EddyCMWF** [comment]
> 

My thinking was to improve performance by only importing modules that if they are used, but happy to put the imports back to the top of the function.

(Side note, `black` made my code non-PEP8 compliant \_O_/)

**Comment by alexamici** [comment]
@EddyCMWF 

> My thinking was to improve performance by only importing modules if they are used, but happy to put the imports back to the top of the function.

My rule is: "readability over performance, unless there is a benchmark" 😄

**Review by alexamici — APPROVED** [review]

**Comment by EddyCMWF** [comment]
@alexamici , sorry to chage after your approval, I realised this should have a test in place, so have just added one

**Comment by iainrussell** [comment]
Hi @EddyCMWF, thanks for adding the tests! I'm happy to merge this into master if you're finished now.

**Comment by EddyCMWF** [comment]
Hi @iainrussell , yes I am finished now

**Pull request merged**
